app/services/helpers/reduction-helper.js

```diff
@@ -1,24 +1,30 @@
 
 var dateFormatter = require('../date-formatter')
+var reductionStatusType = require('../../constants/reduction-status-type')
 
 module.exports.getReductionsByStatus = function (reductions) {
   var active = []
   var scheduled = []
   var archived = []
   var currentDate = new Date()
-  reductions.filter(function (reduction) {
-    if ((reduction.status !== 'ARCHIVED') && (reduction.status !== 'DELETED') &&
-        (currentDate.getTime() > reduction.reductionStartDate.getTime()) &&
-        (currentDate.getTime() < reduction.reductionEndDate.getTime())) {
-      active.push(reduction)
-    } else if ((reduction.status !== 'ARCHIVED') && (reduction.status !== 'DELETED') &&
-      (currentDate.getTime() < reduction.reductionStartDate.getTime())) {
-      scheduled.push(reduction)
-    } else if (reduction.status === 'ARCHIVED') {
-      archived.push(reduction)
-    }
-    formatReductionDates(reduction)
-  })
+  if (reductions instanceof Array) {
+    reductions.filter(function (reduction) {
+      if ((reduction.reductionStartDate instanceof Date) &&
+        (reduction.reductionEndDate instanceof Date)) {
+        if ((reduction.status !== reductionStatusType.ARCHIVED) && (reduction.status !== reductionStatusType.DELETED) &&
+          (currentDate.getTime() > reduction.reductionStartDate.getTime()) &&
+          (currentDate.getTime() < reduction.reductionEndDate.getTime())) {
```

Comment:
Can we just use a check for reductionStatusType.SCHEDULED / ACTIVE here instead of the time range check, as the worker should ensure all statuses are correct in the DB?

Reply:
I would have been more comfortable to remove the date checks if the status attribute was not nullable. However we are not able to set the status to not nullable. As we rely on the reduction worker to set the status of a reduction I am ok relying on that and just check the status and not use the dates in the helper.